cloud/services/scalesets/vmss.go

```diff
@@ -193,8 +193,25 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		}
 	}
 
+	// Assign Identity to VMSS
+	if vmssSpec.Identity == infrav1.VMIdentitySystemAssigned {
+		vmss.Identity = &compute.VirtualMachineScaleSetIdentity{
+			Type: compute.ResourceIdentityTypeSystemAssigned,
```

Comment:
for system assigned, where is the role assignment being reconciled for AzureMachinePools? similar to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/controllers/azuremachine_reconciler.go#L82 for AzureMachines

Reply:
I didn't call this out b/c I was assuming the identity has been created with the roles needed before being used. I think this indicates that we should be specific about the expectations here.

Reply:
That's for user assigned. For system assigned the identity is tied to the VM's lifecycle so it can't be created ahead of time by the user no?

Reply:
Oops! You are totally right.

Reply:
Thanks @CecileRobertMichon , I think I missed the role assignment part for system assigned identity. I'll take a look into it.